components/Timeline.tsx

```diff
@@ -0,0 +1,43 @@
+// *Imports
+import React from "react";
+import { Text, View } from "react-native";
+
+// *Constants
+const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+const today = new Date();
+const todayIndex = today.getDay();
+
+// *Start of the Timeline Function
+export default function Timeline() {
+  const firstDayOfWeek = new Date(today);
+  firstDayOfWeek.setDate(today.getDate() - todayIndex);
+
+  return (
+    <View className="flex-row space-x-3 items-center">
+      {days.map((day, index) => {
+        const dayDate = new Date(firstDayOfWeek);
+        dayDate.setDate(firstDayOfWeek.getDate() + index);
+        const dateNumber = dayDate.getDate();
+
+        return (
+          <View key={index} className="flex-col items-center">
+            <Text className="mb-2 font-semibold">{day}</Text>
+            <View
+              className={`w-10 h-10 rounded-full shadow-md shadow-gray-200 flex items-center justify-center ${
+                todayIndex === index ? "border border-[#5731D6]" : "bg-white"
+              }`}
+            >
+              <Text
```

Comment:
```
// *Imports
import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { LinearGradient } from 'expo-linear-gradient';
import MaskedView from "@react-native-masked-view/masked-view";

// *Constants
const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
const today = new Date();
const todayIndex = today.getDay();

// *Start of the Timeline Function
export default function Timeline() {
  const firstDayOfWeek = new Date(today);
  firstDayOfWeek.setDate(today.getDate() - todayIndex);

  return (
    <View style={styles.container}>
      {days.map((day, index) => {
        const dayDate = new Date(firstDayOfWeek);
        dayDate.setDate(firstDayOfWeek.getDate() + index);
        const dateNumber = dayDate.getDate();

        return (
          <View key={index} style={styles.dayContainer}>
            <Text style={styles.dayText}>{day}</Text>
            <View
              style={[
                styles.circle,
                todayIndex === index && styles.activeBorder,
              ]}
            >
              {todayIndex === index ? (
                <MaskedView
                  style={styles.maskedView}
                  maskElement={
                    <View style={styles.centerText}>
                      <Text style={styles.maskedText}>{dateNumber}</Text>
                    </View>
                  }
                >
                  <LinearGradient
                    colors={["#D372E5", "#5731D6"]}
                    start={{ x: 0, y: 0 }}
                    end={{ x: 1, y: 1 }}
                    style={styles.gradient}
                  />
                </MaskedView>
              ) : (
                <Text style={styles.defaultText}>{dateNumber}</Text>
              )}
            </View>
          </View>
        );
      })}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    gap: 12,
    // justifyContent: "space-around",
  },
  dayContainer: {
    flexDirection: "column",
    alignItems: "center",
  },
  dayText: {
    marginBottom: 8,
    fontWeight: "600",
  },
  circle: {
    width: 40,
    height: 40,
    borderRadius: 20,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
    shadowColor: "#gray",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 2,
  },
  activeBorder: {
    borderWidth: 2,
    borderColor: "#5731D6",
  },
  maskedView: {
    width: "100%",
    height: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  gradient: {
    width: 40,
    height: 40,
  },
  centerText: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  maskedText: {
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "black",
  },
  defaultText: {
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "black",
  },
});
``` 
I found something called MaskedView, which would allow us to have gradient text. For the border, you could probably just add a linear gradient circle behind the selected circle. Here's some example code for the timeline with just the text change. Not sure if you still want to pursue gradient colored text, but it might be worth looking into if we ever have gradient text components/UI in the future.

Reply:
I just noticed that there does seem to be some sort of thing where the circle becomes transparent for the selected day in the little code snippet, but hopefully when actually used, we could try to fix that